### PR TITLE
#171-Rating of books

### LINKED ISSUE #171: Rating of books
Rewrite the algorithm for rating books

### PR DESCRIPTION
**Pull request by Aarrtteemm123**
Rewrite the algorithm for rating books

### CODE REVIEW

File: library/catalog/views.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import random
```
Comment (Palusha) [inline library/catalog/views.py]: Unused import

File: library/catalog/models.py [hunk 2]
```diff
@@ -111,6 +114,28 @@ class Book(Document):
...
+        The expression is the lower bound of a normal approximation to a Bayesian credible interval
+        for the average rating
+        http://www.evanmiller.org/ranking-items-with-star-ratings.html
+
+        """
+        N = sum(self.statistic.stars)
+        K = len(self.statistic.stars)
+        s = list(range(K, 0, -1))
+        s2 = [sk ** 2 for sk in s]
+        z = 1.65
```
Comment (Palusha) [inline library/catalog/models.py]: I think we need more clear name understanding

File: library/catalog/models.py [hunk 3]
```diff
@@ -111,6 +114,28 @@ class Book(Document):
     store_links = ListField(default=[])
     statistic = EmbeddedDocumentField(BookStatistic.__name__, default=BookStatistic())
 
+    def calculate_rating(self):
```
Comment (ivan-dubei) [inline library/catalog/models.py:117]: Yes, we need a better names for variables here.
And where did you get the rating calculation approach (formula) ?

Comment (Aarrtteemm123) [reply library/catalog/models.py:117]: I wrote in doc function a short description and link to the source.

https://www.evanmiller.org/ranking-items-with-star-ratings.html

https://coderoad.ru/1411199/%D0%A7%D1%82%D0%BE-%D0%BC%D0%BE%D0%B6%D0%B5%D1%82-%D0%B1%D1%8B%D1%82%D1%8C-%D0%BB%D1%83%D1%87%D1%88%D0%B5-%D0%B4%D0%BB%D1%8F-%D1%81%D0%BE%D1%80%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%BA%D0%B8-%D0%BF%D0%BE-5-%D0%B7%D0%B2%D0%B5%D0%B7%D0%B4%D0%BE%D1%87%D0%BD%D0%BE%D0%BC%D1%83-%D1%80%D0%B5%D0%B9%D1%82%D0%B8%D0%BD%D0%B3%D1%83

**Review by Palusha — APPROVED** [review]

**Review by ivan-dubei — APPROVED** [review]

**Pull request merged**